build.cake

```diff
@@ -349,9 +352,14 @@ Task("BuildMain")
 Task("BuildTest")
     .IsDependentOn("Setup")
     .IsDependentOn("Restore")
-    .IsDependentOn("BuildMain")
     .Does(() =>
 {
+    foreach (var project in buildPlan.HostProjects)
+    {
+        var projectName = project + ".csproj";
+        var projectFilePath = CombinePaths(env.Folders.Source, project, projectName);
+        BuildProject(env, projectName, projectFilePath, configuration, "test");
+    }
```

Comment:
This is kind of weird. The `BuildTest` task shouldn't build the host projects. IMO, it should just depend on `BuildMain` (which should be renamed to `BuildHosts`).

Reply:
I had to do this because building the hosts was emitting an exe for both projects, which then running unit tests would fail, because it could not find the dll.  It was the best solution I could think of at the time anyway.

Reply:
I'm not following. What you're doing here should be functionally the same as depending on `BuildMain`.

Reply:
OK. I see it now. You're trying to set this "TestOutputType" property as an environment variable. I get why you added this, but it feels pretty messy.

I would argue that OmniSharp.Http and OmniSharp.Stdio should just be super slim exes. The functionality that needs testing should go into libraries. Or, maybe better, we could just have a single OmniSharp.exe that dynamically loads OmniSharp.Http or OmniSharp.Stdio depending on command line args? What do you think?